core/chain_manager_test.go

```diff
@@ -483,22 +485,71 @@ func TestInsertNonceError(t *testing.T) {
 	}
 }
 
-/*
-func TestGenesisMismatch(t *testing.T) {
-	db, _ := ethdb.NewMemDatabase()
-	var mux event.TypeMux
-	genesis := GenesisBlock(0, db)
-	_, err := NewChainManager(genesis, db, db, db, thePow(), &mux)
-	if err != nil {
+func TestChainReorgMissingTransactions(t *testing.T) {
```

Comment:
The point is not so much to have the same transaction inserted in to the chain, rather it shouldn't. The test is to make sure the transaction is removed from the database.